src/code/z_camera.c

```diff
@@ -5252,19 +5252,19 @@ s32 Camera_Unique9(Camera* camera) {
             rwData->curKeyFrame = &ONEPOINT_CS_INFO(camera)->keyFrames[rwData->curKeyFrameIdx];
             rwData->keyFrameTimer = rwData->curKeyFrame->timerInit;
 
-            if (rwData->curKeyFrame->field != ONEPOINT_CS_FIELD_TYPE_NONE) {
-                if ((rwData->curKeyFrame->field & ONEPOINT_CS_FIELD_TYPE_MASK) == ONEPOINT_CS_FIELD_TYPE_ACTORCAT) {
-                    D_8011D3AC = rwData->curKeyFrame->field & 0xF;
-                } else if ((rwData->curKeyFrame->field & ONEPOINT_CS_FIELD_TYPE_MASK) ==
-                           ONEPOINT_CS_FIELD_TYPE_HUD_VISIBILITY) {
-                    Camera_UpdateInterface(CAM_INTERFACE_FIELD(CAM_LETTERBOX_IGNORE, rwData->curKeyFrame->field, 0));
+            if (rwData->curKeyFrame->initField != ONEPOINT_CS_INIT_FIELD_NONE) {
+                if (ONEPOINT_CS_INIT_FIELD_IS_TYPE_ACTORCAT(rwData->curKeyFrame->initField)) {
+                    D_8011D3AC = rwData->curKeyFrame->initField & 0xF;
+                } else if (ONEPOINT_CS_INIT_FIELD_IS_TYPE_HUD_VISIBILITY(rwData->curKeyFrame->initField)) {
+                    Camera_UpdateInterface(
+                        CAM_INTERFACE_FIELD(CAM_LETTERBOX_IGNORE, rwData->curKeyFrame->initField, 0));
                 } else if ((camera->player->stateFlags1 & PLAYER_STATE1_27) &&
                            (player->currentBoots != PLAYER_BOOTS_IRON)) {
                     func_8002DF38(camera->play, camera->target, PLAYER_CSMODE_8);
                     osSyncPrintf("camera: demo: player demo set WAIT\n");
                 } else {
-                    osSyncPrintf("camera: demo: player demo set %d\n", rwData->curKeyFrame->field);
-                    func_8002DF38(camera->play, camera->target, rwData->curKeyFrame->field);
+                    osSyncPrintf("camera: demo: player demo set %d\n", rwData->curKeyFrame->initField);
+                    func_8002DF38(camera->play, camera->target, rwData->curKeyFrame->initField);
                 }
```

Comment:
A comment like "initField indicates a player csMode" or something would be nice

Furthermore, I would unmerge this if else and wrap it inside the else like:
```
if actorcat {
  ...
} else if hudvisibility {
  ...
} else { // playercsmode
  if (random check) {
    csMode = PLAYER_CSMODE_8
  } else {
    csMode = initField
  }
}
```

But I don't care big time so if you don't like it feel free to ignore